shared/js/https.js

```diff
@@ -1,228 +1,211 @@
+                    }
 
-            // Only deal with http calls
-            const protocol = utils.getProtocol(reqUrl).toLowerCase()
-            if (!protocol.indexOf('http:') === 0) {
-                if (this.isSync) { return reqUrl } else { return resolve(reqUrl) }
-            }
+                    try {
+                        let parsedList = JSON.parse(results['https-upgrade-list'])
+                        resolve(parsedList)
+                    } catch(e) {
+                        console.log("HTTPS: getFromStorage() error parsing JSON from chrome.storage.local", e)
+                        reject()
+                    }
+                })
 
-            // Obey global settings (options page)
-            if (!settings.getSetting('httpsEverywhereEnabled')) {
-                if (this.isSync) { return reqUrl } else { resolve(reqUrl) }
-            }
+            // For Safari
+            } else if (window.localStorage) {
+                console.log("HTTPS: getFromStorage() using localStorage (Safari)")
 
-            // Skip upgrading sites that have been whitelisted by user
-            // via on/off toggle in popup
-            if (tab.site.whitelisted) {
-                console.log(`HTTPS: ${tab.site.domain} was whitelisted by user. skip upgrade check.`)
-                if (this.isSync) { return reqUrl } else { return resolve(reqUrl) }
-            }
+                if (!window.localStorage || !localStorage['https-upgrade-list0']) {
+                    return reject()
+                }
 
-            // Skip upgrading sites that have been 'HTTPSwhitelisted'
-            // bc they contain mixed https content when forced to upgrade
-            if (tab.site.HTTPSwhitelisted) {
-                console.log(`HTTPS: ${tab.site.domain} has known mixed content. skip upgrade check.`)
-                if (this.isSync) { return reqUrl } else { return resolve(reqUrl) }
-            }
+                let data = ''
+                for (let i=0; i<LOCAL_STORAGE_CHUNKS; i++) {
+                    data += localStorage['https-upgrade-list' + i]
+                }
 
-            // If `isMainFrame` request and host has known mixed content,
-            // skip db check (don't force upgrade)
-            if (isMainFrame) {
-                if (knownMixedContentList && knownMixedContentList[tab.site.domain]) {
-                    console.log(`HTTPS: ${tab.site.domain} has known mixed content. skip upgrade check.`)
-                    if (this.isSync) { return reqUrl } else { return resolve(reqUrl) }
+                try {
+                    let parsedList = JSON.parse(data)
+                    resolve(parsedList)
+                } catch(e) {
+                    console.log("HTTPS: getFromStorage() error parsing JSON from localStorage", e)
+                    reject()
                 }
+            } else {
+                reject()
             }
+        })
+    }
+
+    saveToStorage (data) {
```

Comment:
I'm fine with parameterizing this, but I don't like saving a data object to a hardcoded key. Put another way, I can associate any data with a specific key here. It's not likely; but it's still not my favorite interface.

Reply:
I was trying to encapsulate that in Chrome/FF it's one entry, in Safari it has to be chunked to multiple key/val entries...

Reply:
I think we're talking across each other here, but maybe not. I'm saying `saveToStorage` should either be without parameter, and use 2 hardcoded / instance variables in the function, or have 2 key, value.

```JavaScript
// This jawn saves the in memory rules set to local storage.
saveToStorage()

//This jawn is a generic interface to save stuff to local storage
saveToStorage(localStorageKey, localStorageData)
```

Perhaps I'm being pedantic though.